### ISSUE TITLE
Physically Based Rendering glTF2 extensions support for for Clear Coat, Transmission and Sheen

### ISSUE THREAD

**Issue by fire** [godotengine/godot-proposals#1934]
### Describe the project you are working on
Godot Engine

### Describe the problem or limitation you are having in your project
We want to import objects that have support a larger range of material from real life.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

This news was mentioned by @akien-mga in developer chat.

Implement Physically Based Rendering glTF2 extensions support for for Clear Coat, Transmission and Sheen.

https://www.khronos.org/news/press/khronos-releases-wave-of-new-gltf-pbr-3d-material-capabilities

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

1. Clear coat already exists in Godot Engine.
2. Transmission already exists in Godot Engine.
3. Sheen is not implemented.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Not without writing a gltf2 importer.

### Is there a reason why this should be core and not an add-on in the asset library?
Gltf2 is core code.

**Comment by huwpascoe** [comment]
For proper cloth materials, sheen is very much needed. The current "rim lighting" option isn't fit for anything, really.

Full spec is here with implementation details
https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_sheen

At minimum should provide
sheenColorFactor
sheenRoughnessFactor

Nice to have
sheenColorTexture
sheenRoughnessTexture

**Comment by Calinou** [comment]
> For proper cloth materials, sheen is very much needed. The current "rim lighting" option isn't fit for anything, really.

What's the difference between glTF sheen and rim lighting? As I understand it, it offers very similar parameters to what Rim has now.

I find Rim works well for cartoony visuals (when you want certain objects to stand out), but one issue is that you can't adjust rim roughness separately from material roughness. This means that for rough materials, the rim effect will often brighten the entire object, not just its edges relative to the current camera perspective. You need to decrease material roughness for this to happen, and this will make the material shiny, which is not always desired.

I think adding a roughness scale parameter to Rim would improve this a lot. To fully comply with the glTF specification, we could also have a custom sheen color parameter, as right now, you can only blend between the light color and the albedo color (using the Tint parameter).

Rim is implemented in the scene shader, so this can't be done with a custom shader unless you reimplement it from scratch in a custom shader.

**Comment by huwpascoe** [comment]
> the rim effect will often brighten the entire object, not just its edges relative to the current camera perspective

Exactly! It doesn't conserve energy, it doesn't interact with shadows, it's not PBR. It could be called fresnel_emission.

![Image](https://github.com/user-attachments/assets/e94510fa-f189-411c-8347-dfec090efd40)

Except it can't, it doesn't use fresnel. The other parts of the shader use the SchlickFresnel function, F0, etc. consistently. The rim light instead uses... whatever this is.
```
// Epsilon min to prevent pow(0, 0) singularity which results in undefined behavior.
half rim_light = pow(max(half(1e-4), half(1.0) - cNdotV), max(half(0.0), (half(1.0) - roughness) * half(16.0)));
```

It's like an unchecked wart stuck to the standard shader.

**Comment by huwpascoe** [comment]
Prototype with direct lighting. I'll bring this up in devchat sometime.
<img width="770" height="562" alt="Image" src="https://github.com/user-attachments/assets/c547068b-2f47-4182-b9a3-0c88a5cc7887" />
